anchore_engine/services/catalog/catalog_impl.py

```diff
@@ -1964,7 +1968,9 @@ def add_or_update_image(
             dockerfile = None
             dockerfile_mode = None
 
-    # logger.debug("rationalized input for imageId ("+str(imageId)+"): " + json.dumps(image_ids, indent=4))
+    # Get a catalog client, for any events that need to be emitted
+    catalog_client = internal_client_for(CatalogClient, userId)
```

Comment:
This logic is already executing in the catalog service, so the client is not needed for invoking ops with in the service. Service clients are used for communicating between services via internal APIs. Using the catalog client may work but essentially the request leaves the service, gets routed over the wire and comes back to the same service (not instance) which is unnecessary. Use add_event() in this module to create a new event